NBitcoin/TransactionBuilder.cs

```diff
@@ -700,6 +702,24 @@ public TransactionBuilder Send(IDestination destination, Money amount)
 			return Send(destination.ScriptPubKey, amount);
 		}
 
+		/// <summary>
+		/// Send all coins added so far with no change. It will automatically estimate the fee.
+		/// </summary>
+		/// <param name="destination"></param>
+		/// <returns></returns>
+		public TransactionBuilder SendAll(IDestination destination, FeeRate rate)
+			=> SendAll(destination.ScriptPubKey, rate);
+
+		public TransactionBuilder SendAll(Script scriptPubKey, FeeRate rate)

```

Comment:
I think it's better to make SendAll() not receive FeeRate param (in the same way that other Send methods don't receive it). You can call EstimateFee() from the test after SendAll(), and use the result of EstimateFee() with SendFees() after that.

Reply:
You are right. I once thought that doesn't work. But it did. (Maybe because I was not aware of how to use `SubtractFee()` method)